llm/llama.go

```diff
@@ -354,14 +438,28 @@ func newLlama(model string, adapters []string, runner ModelRunner, opts api.Opti
 
 		llm := &llama{Options: opts, Running: Running{Port: port, Cmd: cmd, Cancel: cancel}}
 
-		log.Print("starting llama.cpp server")
+		log.Print("starting llama runner")
 		if err := llm.Cmd.Start(); err != nil {
-			log.Printf("error starting the external llama.cpp server: %v", err)
+			log.Printf("error starting the external llama runner: %v", err)
 			continue
 		}
 
+		// monitor the command, it is blocking, so if it exits we need to capture that
+		go func() {
+			err := llm.Cmd.Wait() // this will block until the command exits
```

Comment:
I'm not a fan of this structurally but I can't think of an alternative. That suggests this code needs to be refactored, perhaps separating the subprocess runner from the llm

Reply:
I agree with this, lets do it in a follow-up change to keep things clean in my opinion.